### PR TITLE
chore(ds): use the latest ark-ui/react

### PR DESCRIPTION
**Pull request by IzumiSy**
# Background

We have rolled back the version of @chakra-ui/ark in #111, but would like to use the latest one if possible.

# Changes

* Fix versions of @ark-ui/react and @ark-ui/anatomy.
* Use subpath exports in components that depend on ark-ui.
* Removed unnecessary `use client` banners.

### CODE REVIEW

File: packages/design-systems/src/components/Textarea.tsx
Comment (IzumiSy) [inline packages/design-systems/src/components/Textarea.tsx:1]: This component is already implemented in `components/Text.tsx`, so removed duplication.

File: packages/design-systems/src/components/composite/Datagrid/ColumnFeature/PinnedColumn.tsx [hunk 1]
```diff
@@ -1,5 +1,3 @@
-"use client";
-
```
Comment (IzumiSy) [inline packages/design-systems/src/components/composite/Datagrid/ColumnFeature/PinnedColumn.tsx:2]: `use client` banners in each component are not necessary anymore. 

tsup will add it in bundling instead.

File: packages/design-systems/package.json [hunk 2]
```diff
@@ -71,8 +72,8 @@
     "zod": "^3.22.4"
   },
   "dependencies": {
-    "@ark-ui/anatomy": "^2.2.0",
-    "@ark-ui/react": "~2.0.2",
+    "@ark-ui/anatomy": "2.3.1",
+    "@ark-ui/react": "2.2.2",
```
Comment (IzumiSy) [inline packages/design-systems/package.json:76]: Stricter versions

File: packages/design-systems/src/components/Avatar.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import {
   AvatarFallback,
   AvatarImage,
   AvatarRootProps as AvatarArkProps,
-} from "@ark-ui/react";
+} from "@ark-ui/react/avatar";
```
Comment (IzumiSy) [inline packages/design-systems/src/components/Avatar.tsx:6]: Due to the bug in ark-ui/react (https://github.com/chakra-ui/ark/issues/2266), all components are importing ark-ui components with subpaths.

File: packages/design-systems/tsup.config.ts [hunk 4]
```diff
@@ -1,4 +1,22 @@
...
+      }
+    : {};
+
+const commonOpts: Options = {
+  format: ["cjs", "esm"],
+  external: ["react"],
+  dts: true,
+  clean: true,
+  ...devOpts,
+};
```
Comment (IzumiSy) [inline packages/design-systems/tsup.config.ts:19]: Commonized build configuration

File: packages/design-systems/src/components/Textarea.tsx
Comment (IzumiSy) [inline packages/design-systems/src/components/Textarea.tsx:1]: This component is already implemented in components/Text.tsx, so removed duplication.

Comment (IzumiSy) [inline packages/design-systems/src/components/Textarea.tsx:1]: This component is already implemented in components/Text.tsx, so removed duplication.

**Review by murayama-r — APPROVED** [review]
LGTM

**Pull request merged**